### PR TITLE
Introduce `ValueConversionException`

### PR DESCRIPTION
**Pull request by yhabteab**

### CODE REVIEW

**Review by nilmerg — APPROVED** [review]

**Comment by lippserd** [comment]
@nilmerg I don't like `BadInputException` because the name is too general for such a specific use case. Do you perhaps have a suggestion? I like something like `ValueConversionException`.

**Comment by nilmerg** [comment]
Not really.

File: src/Compat/FilterProcessor.php [hunk 1]
```diff
@@ -112,11 +113,16 @@ protected function requireAndResolveFilterColumns(Filter\Rule $filter, Query $qu
                 $subjectBehaviors = $resolver->getBehaviors($subject);
 
                 // Prepare filter as if it were final to allow full control for rewrite filter behaviors
-                $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
                 $filter->setColumn($resolver->getAlias($subject) . '.' . $columnName);
                 $filter->metaData()->set('columnName', $columnName);
                 $filter->metaData()->set('relationPath', $path);
 
+                try {
+                    $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
```
Comment (lippserd) [inline src/Compat/FilterProcessor.php:121]: Either the comment should also be moved here or the try-catch block should stay in the same place as the line it replaces. I would prefer the latter.

File: src/Compat/FilterProcessor.php [hunk 2]
```diff
@@ -111,8 +112,14 @@ protected function requireAndResolveFilterColumns(Filter\Rule $filter, Query $qu
 
                 $subjectBehaviors = $resolver->getBehaviors($subject);
 
-                // Prepare filter as if it were final to allow full control for rewrite filter behaviors
-                $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
+                try {
+                    // Prepare filter as if it were final to allow full control for rewrite filter behaviors
+                    $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
+                } catch (ValueConversionException $_) {
+                    // The search bar may submit values with wildcards or whatever the user is inputting.
```
Comment (lippserd) [inline src/Compat/FilterProcessor.php:119]: Please write `has entered` instead of `is inputting`.

**Review by lippserd — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Review by lippserd — APPROVED** [review]

**Pull request merged**